xmtp_mls/src/groups/message_history.rs

```diff
@@ -178,37 +272,158 @@ where
         Ok(())
     }
 
-    pub(crate) fn provide_pin(&self, pin_challenge: &str) -> Result<(), GroupError> {
-        let conn = self.store().conn()?;
-        let sync_group_id = conn
-            .find_sync_groups()?
-            .pop()
-            .ok_or(GroupError::GroupNotFound)?
-            .id;
+    pub async fn get_pending_history_request(
```

Comment:
I'm OK with having this not support there being more than one pending history request for now, but we should probably think through how we can support that down the line.